torch/distributed/_functional_collectives.py

```diff
@@ -118,13 +118,17 @@ def cleanup(self):
                 del data_ptr_to_work[ptr]
 
 
-def _register_tensor_work(tensor_or_list, work):
-    reg = _WaitRegistration(work)
+def _register_tensor_work(tensor_or_list, work_or_list):
     if not isinstance(tensor_or_list, list):
```

Comment:
should we define the api better? maybe assert that work_or_list size matches tensor_or_list size, or that work is a list implies tensor is also a list?